main/SQLHelper.cpp

```diff
@@ -7529,6 +7529,30 @@ void CSQLHelper::DeleteScenes(const std::string& idx)
 	m_notifications.ReloadNotifications();
 }
 
+void CSQLHelper::CopyNewDevice(const std::string& newidx, const std::string& idx)
+{
+	// This function copy fields describing hardware from the new device to the old.
+	safe_query("UPDATE DeviceStatus set (DeviceID, Unit, Type, SubType) = (select DeviceID, Unit, Type, SubType from DeviceStatus d2 where d2.ID = '%q') where ID='%q'" , newidx.c_str(), idx.c_str());
+}
+
+// tranfer logs from idx to newidx for table "table"
+void CSQLHelper::Transfer1Log(const char* table,const std::string& idx, const std::string& newidx)
+{
```

Comment:
I'm not quite sure what happens here...
Not sure if deleting log records from the new device is wanted.
(My new device could be running for some days before I want to exchange it)
But sure if we transfer from A to B, log records for B should not overlap A, so that is correct to delete them (B)

After this, the only thing needed to be done is set the DeviceRowID from B to A and update the DeviceStatus (CopyNewDevice)

May I suggest to rename 'CopyNewDevice' to 'TransferDevice' and as parameters 'old_idx' and 'new_idx" to make it clearer
And rename 'Transfer1Log' to 'TransferLog' (without the 1) and also 'old_idx' and 'new_idx"



I don't think we need extreme dates, you just have to update the DeviceRowID